lib/Gateway.js

```diff
@@ -466,6 +466,22 @@ function getMappedValuesTemplate (valueId, modeMap, integerList, defaultValue) {
   return `{{ ${JSON.stringify(finalMap)}[value_json.value${valueFilter}] | default('${defaultValue}') }}`
 }
 
+/**
+ * Retrives the value of a property from the node valueId
+ *
+ * @param {Object} payload discovery payload
+ * @param {String} prop property name
+ * @param {Object} node node object
+ */
+function setDiscoveryValue (payload, prop, node) {
+  if (typeof payload[prop] === 'string') {
+    const valueId = node.values[payload[prop]]
+    if (valueId && valueId.value != null) {
```

Comment:
`!==`

Reply:
I have used `!=` null as it also works with `undefined`

Reply:
i'm still not sure about that, what other types are we anticipating it being?

Reply:
What are you not sure about? Anyway we could make it !== and it should work anyway as I don't know when and why value could be undefined

Reply:
you're relying on typecasting, i'm not sure I would grok it that way you're describing.

a more obvious (to me at least) way of expressing the same is:
```js
if (valueId && valueId.value)
```
though given we don't know the reprocrussions for other types maybe it would be better to check if its a string/number. if an object/array/etc snuck in there I imagine it'd break other things?

Reply:
@chrisns What if `valueId.value` is 0? Your check would fail

Reply:
yes, i should have actually drank my ☕  before writing that!
ignore me, leave it with `!=`